### ISSUE TITLE
📊 TradingView-Compatible RSI Calculator (Wilder's Smoothing)

### ISSUE THREAD

**Issue by abhidp** [abhidp/quant-trading-engine#59]
**Priority:** HIGH  
**Estimate:** 3-4 hours  

## Mission
Implement TradingView-compatible RSI calculator using Wilder's Smoothing (RMA) method. DO NOT replace existing RSI logic - keep as legacy for fallback.

## CRITICAL REQUIREMENT: PRESERVE EXISTING CODE
- **DO NOT DELETE** existing RSI calculator
- **RENAME** existing RSICalculator to RSICalculatorLegacy
- **CREATE NEW** TradingViewRSICalculator class
- **MAINTAIN BACKWARD COMPATIBILITY** with existing strategy

## Problem Analysis
Current Python RSI uses Simple Moving Average (SMA) for smoothing:
```python
gain = (delta.where(delta > 0, 0)).rolling(window=period).mean()  # SMA
loss = (-delta.where(delta < 0, 0)).rolling(window=period).mean()  # SMA
```

TradingView RSI uses Wilder's Smoothing (RMA):
```pinescript
up = ta.rma(math.max(change, 0), rsiLengthInput)    # RMA (Wilder's)
down = ta.rma(-math.min(change, 0), rsiLengthInput)  # RMA (Wilder's)
```

This creates different RSI values and affects signal accuracy.

## Acceptance Criteria
- [ ] Rename existing RSICalculator to RSICalculatorLegacy in indicators.py
- [ ] Implement TradingViewRSICalculator class using Wilder's Smoothing (RMA)
- [ ] Add RSI-MA crossover signal generation functionality
- [ ] Include validation method to compare against TradingView values
- [ ] Maintain exact same interface as legacy RSI for easy switching
- [ ] Add comprehensive unit tests for both legacy and TradingView RSI
- [ ] Update strategy to use TradingView RSI by default but allow legacy fallback
- [ ] Document differences between legacy and TradingView calculations

### Core RSI Calculation (Match TradingView exactly)
```python
class TradingViewRSICalculator:
    """
    TradingView-compatible RSI using Wilder's Smoothing (RMA)
    Matches TradingView's default RSI indicator exactly
    """
    
    def __init__(self, period=14):
        self.period = period
    
    def _rma(self, series, period):
        """
        Wilder's Smoothing (RMA) - exact equivalent of TradingView's ta.rma()
        Formula: alpha = 1/period, then exponential smoothing
        """
        alpha = 1.0 / period
        return series.ewm(alpha=alpha, adjust=False).mean()
    
    def calculate(self, prices, period=None):
        """
        Calculate RSI using Wilder's Smoothing (matches TradingView)
        
        TradingView Pine Script equivalent:
        change = ta.change(close)
        up = ta.rma(math.max(change, 0), 14)
        down = ta.rma(-math.min(change, 0), 14)
        rsi = down == 0 ? 100 : up == 0 ? 0 : 100 - (100 / (1 + up / down))
        """
        if period is None:
            period = self.period
            
        # Step 1: Calculate price changes (ta.change equivalent)
        change = prices.diff()
        
        # Step 2: Separate gains and losses (math.max/min equivalent)
        gains = change.where(change > 0, 0)
        losses = -change.where(change < 0, 0)
        
        # Step 3: Apply Wilder's smoothing (ta.rma equivalent)
        avg_gains = self._rma(gains, period)
        avg_losses = self._rma(losses, period)
        
        # Step 4: Calculate RSI with TradingView's exact edge case handling
        rs = avg_gains / avg_losses
        rsi = np.where(
            avg_losses == 0, 100,  # down == 0 ? 100
            np.where(
                avg_gains == 0, 0,  # up == 0 ? 0  
                100 - (100 / (1 + rs))  # standard formula
            )
        )
        
        return pd.Series(rsi, index=prices.index, name='rsi')
```

### RSI-MA Crossover Functionality
```python
def calculate_with_ma_crossover(self, prices, ma_period=14, ma_type='SMA'):
    """
    Calculate RSI with Moving Average crossover signals
    Perfect for RSI-MA crossover strategy within 30/70 bands
    
    Signals:
    - BUY: RSI crosses above MA within oversold recovery zone (25-40)
    - SELL: RSI crosses below MA within overbought zone (60-75)
    """
    rsi = self.calculate(prices)
    
    # Calculate RSI moving average (like TradingView smoothing options)
    if ma_type == 'SMA':
        rsi_ma = rsi.rolling(window=ma_period).mean()
    elif ma_type == 'EMA':
        rsi_ma = rsi.ewm(span=ma_period, adjust=False).mean()
    elif ma_type == 'RMA':
        rsi_ma = self._rma(rsi, ma_period)
        
    # Generate crossover signals
    df = pd.DataFrame({'rsi': rsi, 'rsi_ma': rsi_ma})
    df['rsi_cross_above_ma'] = (df['rsi'] > df['rsi_ma']) & (df['rsi'].shift(1) <= df['rsi_ma'].shift(1))
    df['rsi_cross_below_ma'] = (df['rsi'] < df['rsi_ma']) & (df['rsi'].shift(1) >= df['rsi_ma'].shift(1))
    
    # Filtered signals within specific RSI ranges
    df['buy_signal'] = df['rsi_cross_above_ma'] & (df['rsi'] > 25) & (df['rsi'] < 40)
    df['sell_signal'] = df['rsi_cross_below_ma'] & (df['rsi'] > 60) & (df['rsi'] < 75)
    
    return df
```

### Validation Against TradingView
```python
def validate_against_tradingview(self, prices, tradingview_values=None):
    """
    Validate RSI calculation matches TradingView
    
    Usage:
    1. Calculate RSI in TradingView with same data
    2. Export last 10 RSI values from TradingView
    3. Compare with this function output
    """
    rsi = self.calculate(prices)
    recent_rsi = rsi.dropna().tail(10)
    
    validation_result = {
        'latest_rsi': recent_rsi.iloc[-1],
        'recent_values': recent_rsi.tolist(),
        'calculation_method': 'Wilder\'s Smoothing (RMA)',
        'tradingview_compatible': True
    }
    
    if tradingview_values:
        differences = [abs(calc - tv) for calc, tv in zip(recent_rsi, tradingview_values)]
        max_diff = max(differences)
        validation_result['max_difference'] = max_diff
        validation_result['accuracy'] = 'EXCELLENT' if max_diff < 0.1 else 'NEEDS_REVIEW'
    
    return validation_result
```

### Updated indicators.py structure
```python
# indicators.py

# LEGACY RSI (keep for fallback)
class RSICalculatorLegacy:
    """Legacy RSI using Simple Moving Average - keep for fallback"""
    # [existing implementation unchanged]

# NEW TRADINGVIEW RSI (primary)
class TradingViewRSICalculator:
    """TradingView-compatible RSI using Wilder's Smoothing"""
    # [new implementation as specified above]

# Convenience aliases for easy switching
RSICalculator = TradingViewRSICalculator  # Default to TradingView
LegacyRSI = RSICalculatorLegacy          # Explicit legacy access
```

## Strategy Integration Requirements
- [ ] Update signal_generator.py to use TradingViewRSICalculator by default
- [ ] Add configuration option to switch between TradingView and Legacy RSI
- [ ] Ensure backward compatibility - existing strategy should work unchanged
- [ ] Add logging to indicate which RSI calculation method is being used

## Testing Requirements
- [ ] Unit tests comparing Legacy vs TradingView RSI calculations
- [ ] Validation test against actual TradingView RSI values
- [ ] Performance comparison: Legacy RSI strategy vs TradingView RSI strategy
- [ ] RSI-MA crossover signal testing with historical data
- [ ] Ensure no breaking changes to existing codebase

## Configuration Support
```python
# config/strategy_config.py
class StrategyConfig:
    # RSI Calculation Method
    rsi_calculation_method: str = 'tradingview'  # 'tradingview' or 'legacy'
    
    # RSI-MA Crossover Settings (new feature)
    enable_rsi_ma_crossover: bool = False
    rsi_ma_period: int = 14
    rsi_ma_type: str = 'SMA'  # 'SMA', 'EMA', 'RMA'
```

## Expected Benefits
- **Exact TradingView Matching**: RSI values identical to TradingView charts
- **Better Signal Accuracy**: Wilder's smoothing provides more reliable signals
- **New Strategy Options**: RSI-MA crossover within overbought/oversold bands
- **Professional Standard**: Industry-standard RSI calculation method
- **Backward Compatibility**: Can still use legacy RSI if needed

## Validation Process
1. **Calculate RSI in TradingView** with default settings (period=14)
2. **Export RSI values** for same price data 
3. **Run validation function** to compare differences
4. **Ensure differences < 0.1** for accuracy confirmation
5. **Test RSI-MA crossover signals** against TradingView smoothing

## Files to Modify/Create
- [ ] Update `core/indicators.py` (rename existing, add new class)
- [ ] Update `core/signal_generator.py` (use TradingView RSI by default)
- [ ] Create `tests/test_tradingview_rsi.py` (comprehensive testing)
- [ ] Update `config/strategy_config.py` (add RSI method selection)
- [ ] Create `notebooks/rsi_comparison_analysis.ipynb` (compare legacy vs TradingView)

## Implementation Notes
- **Preserve all existing functionality** - zero breaking changes
- **Wilder's Smoothing formula**: alpha = 1/period for exponential smoothing
- **Handle edge cases exactly like TradingView**: zero gains/losses scenarios
- **Maintain same interface** as legacy RSI for seamless switching
- **Add comprehensive logging** for debugging and validation
